utils/httputil/client.go

```diff
@@ -230,10 +234,34 @@ func (c *Client) request(
 			continue
 		}
 
-		if status = r.GetStatus(); status == StatusTooManyRequests || status >= 500 {
+		status = r.GetStatus()
+
+		if status == StatusNirnTimeout || status >= 500 {
 			continue
 		}
 
+		if status == StatusTooManyRequests {
+			var body = r.GetBody()
+			defer body.Close()
+
+			buf := bytes.Buffer{}
+			buf.ReadFrom(body)
+
+			var errBody struct {
+				RetryAfter float32 `json:"retry_after"`
+			}
+
+			err := json.Unmarshal(buf.Bytes(), &errBody)
```

Comment:
We can shorten this to:

```suggestion
			err := json.NewDecoder(body).Decode(&errBody)
```